casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala

```diff
@@ -110,11 +119,15 @@ object BlockApproverProtocol {
       _ <- (blockBonds == bonds)
             .either(())
             .or("Block bonds don't match expected.")
-      validators              = blockBonds.toSeq.map(b => ProofOfStakeValidator(b._1.toByteArray, b._2))
-      genesisBlessedContracts = Genesis.defaultBlessedTerms(timestamp, validators, wallets).toSet
-      blockDeploys            = body.deploys.flatMap(ProcessedDeployUtil.toInternal)
-      genesisBlessedTerms     = genesisBlessedContracts.flatMap(_.term)
-      genesisBlessedDeploys   = genesisBlessedContracts.flatMap(_.raw)
+      validators = blockBonds.toSeq.map(b => ProofOfStakeValidator(b._1.toByteArray, b._2))
+      posParams  = ProofOfStakeParams(minimumBond, maximumBond, validators)
+      faucetCode = if (faucet) Faucet.basicWalletFaucet(_) else Faucet.noopFaucet
+      genesisBlessedContracts = Genesis
+        .defaultBlessedTerms(timestamp, posParams, wallets, faucetCode)
+        .toSet
+      blockDeploys          = body.deploys.flatMap(ProcessedDeployUtil.toInternal)
+      genesisBlessedTerms   = genesisBlessedContracts.flatMap(_.term)
+      genesisBlessedDeploys = genesisBlessedContracts.flatMap(_.raw)
```

Comment:
This feels like a lot of duplicated code. Given how it has the critical boolean flag, it doesn't help to make it a critical boolean flag in two places.

Reply:
But the flag just gets read from the config in both cases. It's not really in two places, it's just referenced in two places. And again this is because this code touches the boundary of the program and so needs to get information from the config.